services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/TormaystarkasteluTulos.kt

```diff
@@ -46,13 +48,59 @@ data class TormaystarkasteluTulos(
             when (max) {
                 linjaautoliikenneindeksi -> IndeksiType.LINJAAUTOLIIKENNEINDEKSI
                 raitioliikenneindeksi -> IndeksiType.RAITIOLIIKENNEINDEKSI
-                autoliikenneindeksi -> IndeksiType.AUTOLIIKENNEINDEKSI
+                autoliikenne.indeksi -> IndeksiType.AUTOLIIKENNEINDEKSI
                 else -> IndeksiType.PYORALIIKENNEINDEKSI
             }
         LiikennehaittaindeksiType(max, type)
     }
 }
 
+@Schema(description = "Car traffic nuisance index and classification")
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class Autoliikenneluokittelu(
+    @JsonView(ChangeLogView::class) val indeksi: Float,
+    @JsonView(ChangeLogView::class) val haitanKesto: Int,
+    @JsonView(ChangeLogView::class) val katuluokka: Int,
+    @JsonView(ChangeLogView::class) val liikennemaara: Int,
+    @JsonView(ChangeLogView::class) val kaistahaitta: Int,
+    @JsonView(ChangeLogView::class) val kaistapituushaitta: Int,
+) {
+    companion object {
+        fun calculateIndeksi(
+            haitanKesto: Int,
+            katuluokka: Int,
+            liikennemaara: Int,
+            kaistahaitta: Int,
+            kaistapituushaitta: Int,
+        ): Float =
+            if (katuluokka == 0 && liikennemaara == 0) {
+                0.0f
+            } else {
+                (0.1f * haitanKesto +
+                        0.2f * katuluokka +
```

Comment:
I'm not really loving having important business logic inside a data class like this. Data classes should be simple and boring.

Maybe the calculation function could be moved to the companion object of `TormaystarkasteluLaskentaService`?

Reply:
Valid point. I moved the function and companion object into `TormaystarkasteluLaskentaService`. And the test. Also, added a separate test for "no street class nor traffic" logic.